core/deployment/src/main/java/io/quarkiverse/operatorsdk/deployment/RBACAugmentationStep.java

```diff
@@ -20,14 +33,60 @@ public boolean getAsBoolean() {
     }
 
     @BuildStep(onlyIf = IsRBACEnabled.class)
-    void augmentRBACForResources(BuildTimeOperatorConfiguration buildTimeConfiguration,
+    void augmentRBACForResources(BuildTimeOperatorConfiguration buildTimeConfiguration, KubernetesConfig kubernetesConfig,
             BuildProducer<DecoratorBuildItem> decorators,
+            BuildProducer<KubernetesClusterRoleBuildItem> clusterRolesProducer,
+            BuildProducer<KubernetesRoleBindingBuildItem> roleBindingsProducer,
+            ApplicationInfoBuildItem applicationInfo,
             ControllerConfigurationsBuildItem configurations) {
 
         final var configs = configurations.getControllerConfigs().values();
-        decorators.produce(new DecoratorBuildItem(
-                new AddClusterRolesDecorator(configs, buildTimeConfiguration.crd().validate())));
-        decorators.produce(new DecoratorBuildItem(
-                new AddRoleBindingsDecorator(configs, buildTimeConfiguration)));
+        final var clusterRoles = new AddClusterRolesDecorator(configs, buildTimeConfiguration.crd().validate())
+                .createClusterRoles();
+        clusterRoles.forEach(clusterRole -> clusterRolesProducer.produce(clusterRoleBuildItemFrom(clusterRole)));
+
+        final var defaultServiceAccountName = ResourceNameUtil.getResourceName(kubernetesConfig, applicationInfo);
+        final var serviceAccountName = kubernetesConfig.getServiceAccount().orElse(defaultServiceAccountName);
+        final var bindings = new AddRoleBindingsDecorator(configs, buildTimeConfiguration, serviceAccountName).createBindings();
+        bindings.forEach(binding -> roleBindingsProducer.produce(roleBindingItemFor(binding)));
+
+        //        decorators.produce(new DecoratorBuildItem(
```

Comment:
forgotten?

Reply:
No, this is a work in progress, I wanted to keep an easy way to revert to the current behavior quickly. Then again, this actually wouldn't work anymore due to another change that is wrong for the current behavior, iirc.